### ISSUE TITLE
dynamodb.query doesn't support FilterExpression

### ISSUE THREAD

**Issue by jakubka** [spulec/moto#715]
Docs:

https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/QueryAndScan.html#FilteringResults

It looks like mock for the `FilterExpression` operator is not implemented. It simply isn't taken into account and everything is always returned.

``` python
table = session.resource('dynamodb').Table('table_name')
result = table.query(KeyConditionExpression=boto3.dynamodb.conditions.Key("key").eq(key),
                     FilterExpression=boto3.dynamodb.conditions.Attr("col").eq(1))
```

This behaves as in `FilterExpression` weren't specified.

Are there plains to add support for `FilterExpression`?

Thanks!

**Comment by pouledodue** [comment]
I need that too

**Comment by atif1996** [comment]
Yeah I need this too.  Seems like it would be simple enough to add.

**Comment by spulec** [comment]
No plans. I would be very happy to accept a pull request if anyone has time to implement.

**Comment by jgrobbel** [comment]
Looks like some basic support is already in (I am looking at scan but query also has some). 
https://github.com/spulec/moto/blob/master/moto/dynamodb2/responses.py#L422
https://github.com/spulec/moto/blob/master/moto/dynamodb2/models.py#L445

`self.body.get('ScanFilter', {})` always returns the default {}. Perhaps that needs hooking up?

**Comment by terrycain** [comment]
Should be able to fix this after #1262

**Comment by JackDanger** [comment]
#1262 is merged, I'll release soon
